### PR TITLE
Job deletion API

### PR DESCRIPTION
**Pull request by nir-sopher**
Allowing the deletion of a Job

### CODE REVIEW

File: traffic_ops/app/lib/TrafficOpsRoutes.pm [hunk 1]
```diff
@@ -531,7 +531,7 @@ sub api_routes {
 	$r->post("/api/$version/deliveryservices/request")->over( authenticated => 1, not_ldap => 1 )->to( 'Deliveryservice#request', namespace => $namespace );
 
 	# -- DELIVERYSERVICES: STEERING DELIVERYSERVICES
-	$r->get("/internal/api/$version/steering")->over( authenticated => 1, not_ldap => 1 )->to( 'Steering#index', namespace => 'API::DeliveryService' );
+	$r->get("/internal/api/$version/steering")->over( authenticated => 0, not_ldap => 0 )->to( 'Steering#index', namespace => 'API::DeliveryService' );
```
Comment (dangogh) [inline traffic_ops/app/lib/TrafficOpsRoutes.pm]: did this change get in by mistake?

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1001/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1002/
Test PASSed.

**Comment by nir-sopher** [comment]
thanks @dangogh 
Fixed

File: traffic_ops/app/lib/API/Job.pm [hunk 2]
```diff
@@ -113,6 +113,29 @@ sub show {
 	$self->success( \@data );
 }
 
+
+sub delete {
```
Comment (mitchell852) [inline traffic_ops/app/lib/API/Job.pm]: I think you need to check tenancy on this method. you can't delete a job outside of your tenancy, right? the job table has a job_deliveryservice column that you can use to check tenancy...

Comment (nir-sopher) [reply traffic_ops/app/lib/API/Job.pm]: Correct, 
it was also missing in the show command.
Need some further testing before pulling

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1009/
Test PASSed.

**Comment by mitchell852** [comment]
i "think" deleting a job is not as simple as removing it from the job table. deleting a job probably needs to set reval_pending = 1 for all servers in the ds's cdn. i think this is what happens when you create a job....otherwise, deleting a job really does  nothing.

**Pull request closed without merging**